src/core/image.js

```diff
@@ -79,6 +79,68 @@ var PDFImage = (function PDFImageClosure() {
     return dest;
   }
 
+  /**
+   * Downsizes an image mask.
+   * @param {TypedArray} src - The source buffer.
+   * @param {Number} w1 - Original width.
+   * @param {Number} h1 - Original height.
+   * @param {Number} w2 - New width.
+   * @param {Number} h2 - New height.
+   * @returns {Object} Downsized image mask buffer.
+   */
+  function downsizeImageMask(src, w1, h1, w2, h2) {
+    let resizeFactorX = Math.round(w1 / w2);
+    let resizeFactorY = Math.round(h1 / h2);
+    if (resizeFactorX === 1 && resizeFactorY === 1) {
+      return null;
+    }
+    let packedWidth = (w1 + 7) >> 3;
+    let xHop = (resizeFactorX >> 3);
+    xHop = xHop ? xHop : 1;
+    let yHop = resizeFactorY;
+    let bitHop;
+    if (resizeFactorX >= 8) {
+      bitHop = 8;
+    } else if (resizeFactorX >= 4) {
+      bitHop = 4;
+    } else if (resizeFactorX >= 2) {
+      bitHop = 2;
+    } else {
+      bitHop = 1;
+    }
+    let destWidth = (Math.ceil(packedWidth / xHop) * (8 / bitHop) + 7) >> 3;
+    let destHeight = Math.ceil(h1 / yHop);
+    let dest = new Uint8Array(destWidth * destHeight);
```

Comment:
This must be `Uint8ClampedArray` instead, since all image handling code has been refactored to return data in that format. (If you rebase the patch against `master`, you'll most likely get even more failures unless the typed array is fixed here.)